### PR TITLE
Log in

### PR DESCRIPTION
**Pull request by kriizjones**
# Pivotal Tracker story

www.pivotaltracker.com/n/projects/1948009

Changes proposed in this pull request:

This feature create the Log in functionality to enable users to log in 

# What I have learned working on this feature:

• recapping TDD and acceptance testing
• installing and setting up Devise
• more about step definitions

Ready for review!

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
-
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
```
Comment (AmberWilkie) [inline config/environments/development.rb]: This line should not be uncommented. I know Devise told you to do it, but we're not using mailers.

File: features/log_in.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Feature: Login to buy and sell an dish
+  As an owner
+  I need to log in
+  So I can sell or buy dishes
+
+  Background:
+    Given that there is a user with email: "kristoffer.user@homechef.com"
+
+
+  Scenario: Visitor logs in so them can create a dish to sell or they can buy a dish
```
Comment (AmberWilkie) [inline features/log_in.feature]: `Scenario: A user logs in`

File: features/log_in.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    When I am on the "landing" page
+    And I click "Login"
+    Then I fill in "Email" with "kristoffer.user@homechef.com"
+    And I fill in "Password" with "password"
+    Then I click "Log in" button
+    And I should be on the "Landing" page
+    And I should see "Signed in successfully."
```
Comment (AmberWilkie) [inline features/log_in.feature:16]:
```
And I click "Log in" button
Then I should be on the "Landing" page
And I should see "Signed in successfully."
```

File: spec/factories/models.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :model do
+    
+  end
+end
```
Comment (AmberWilkie) [inline spec/factories/models.rb]: Delete this file.

**Review by AmberWilkie — CHANGES_REQUESTED** [review]

**Comment by tochman** [comment]
The commit history on this PR is messed up. We need to create feature branches from develop. This needs to be cherry-picked. Not cool.

**Review by tochman — CHANGES_REQUESTED** [review]
Fix commit history

File: config/environments/development.rb (hunk 1 above)
Comment (kriizjones) [reply config/environments/development.rb]: done.

File: features/log_in.feature (hunk 2 above)
Comment (kriizjones) [reply features/log_in.feature]: done.

File: features/log_in.feature (hunk 3 above)
Comment (kriizjones) [reply features/log_in.feature:16]: done . :)

File: spec/factories/models.rb (hunk 4 above)
Comment (kriizjones) [reply spec/factories/models.rb]: done, deleted!

File: features/log_in.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
+Feature: Login to buy and sell an dish
+  As an owner
+  I need to log in
+  So I can sell or buy dishes
```
Comment (tochman) [inline features/log_in.feature]: Please use proper user story format. Update the tracker as well.

File: features/log_in.feature [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+  Background:
+    Given that there is a user with email: "kristoffer.user@homechef.com"
+
+
```
Comment (tochman) [inline features/log_in.feature]: Remove extra line break

File: features/step_definitions/basic_steps.rb [hunk 7]
```diff
@@ -1,3 +1,12 @@
...
   save_and_open_page
+end
+
+When(/^I click "([^"]*)"$/) do |button|
+  click_link_or_button button
+end
+
+Then(/^I fill in "([^"]*)" with "([^"]*)"$/) do |field, text|
+  fill_in field, with: text
+
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Remove extra line break.

**Review by tochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_steps.rb (hunk 7 above)
Comment (kriizjones) [reply features/step_definitions/basic_steps.rb]: done.

File: features/log_in.feature (hunk 6 above)
Comment (kriizjones) [reply features/log_in.feature]: done.

File: features/log_in.feature (hunk 5 above)
Comment (kriizjones) [reply features/log_in.feature]: done.

**Comment by tochman** [comment]
The commit history is a big obstacle for me to be honest. Bit I'm willing to let this go. I will monitor your next PR and I hope you are more on top of things next time.

**Review by tochman — APPROVED** [review]

**Review by AmberWilkie — APPROVED** [review]

**Pull request merged**